pkg/service/auth_context.go

```diff
@@ -63,53 +62,119 @@ func (authContext *AuthContext) evaluateAuthConfig(ctx context.Context, config c
 	}
 }
 
-func (authContext *AuthContext) evaluateAuthConfigs(authConfigs []common.AuthConfigEvaluator, cb configCallback) error {
-	errGroup, ctx := errgroup.WithContext(context.Background())
+func (authContext *AuthContext) evaluateOneAuthConfig(authConfigs []common.AuthConfigEvaluator, cb evaluateCallback) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	waitGroup := new(sync.WaitGroup)
+	size := len(authConfigs)
+	errorChannel := make(chan error, size)
+	successChannel := make(chan bool, size)
+
+	waitGroup.Add(size)
+
+	go func() {
+		waitGroup.Wait()
+		close(errorChannel)
+		close(successChannel)
+	}()
 
 	for _, authConfig := range authConfigs {
 		objConfig := authConfig
-		errGroup.Go(func() error {
-			return authContext.evaluateAuthConfig(ctx, objConfig, cb)
-		})
+		go func() {
+			defer waitGroup.Done()
+
+			err := authContext.evaluateAuthConfig(ctx, objConfig, cb)
+			if err != nil {
+				errorChannel <- err
+			} else {
+				successChannel <- true
+				cancel() // cancels the context if at least one thread succeeds
+			}
+		}()
 	}
-	if err := errGroup.Wait(); err != nil {
-		return err
-	} else {
-		authCxtLog.Info("Successfully fetched all auth objects.", "authConfigs", authConfigs)
+
+	success := <-successChannel
```

Comment:
With this implementation:
* If at least one evaluation succeeds,  other potential errors are not taken into account
* If all evaluations fail, only one err is reported.

This is the expected behavior?

Reply:
This is exactly the expected behaviour and what "OneAuthConfig" is supposed to mean.

Reply:
_Correction:_ more than one err may be reported. However, it doesn't matter. What matters is that if at least one success is reported, it's good enough to continue.